config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/loader/JmsDestinationLoader.java

```diff
@@ -0,0 +1,142 @@
+        final String jndiUrl = (String) jmsConnectionProps.remove(JNDI_PROVIDER_URL);
+
+        final Map<String, Object> contextPropertiesTemplateProps = mapPropertiesElements(getSingleChildElement(jmsConnectionEle, CONTEXT_PROPERTIES_TEMPLATE, false), CONTEXT_PROPERTIES_TEMPLATE);
+        final String jndiUsername = (String) contextPropertiesTemplateProps.remove(JNDI_USERNAME);
+        final String jndiPassword = (String) contextPropertiesTemplateProps.remove(JNDI_PASSWORD);
+        
+        final Map<String, Object> jndiProperties = contextPropertiesTemplateProps.entrySet().stream()
+                .filter(map -> 
+                        !map.getKey().startsWith("com.l7tech.server.jms.prop.") && 
+                        !"com.l7tech.server.jms.soapAction.msgPropName".equals(map.getKey()))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        final String destinationType = (String) jmsDestinationDetailsProps.remove(DESTINATION_TYPE);
+        final String connectionFactoryName = (String) jmsConnectionProps.remove(CONNECTION_FACTORY_NAME);
+        final String destinationName = getSingleChildElementTextContent(jmsDestinationDetailEle, JMS_DESTINATION_NAME);
+        final String destinationUsername = (String) jmsDestinationDetailsProps.remove(PROPERTY_USERNAME);
+        final String destinationPassword = (String) jmsDestinationDetailsProps.remove(PROPERTY_PASSWORD);
+        
+        JmsDestination jmsDestination = new JmsDestination();
+        jmsDestination.setId(id);
+        jmsDestination.setName(name);
+        jmsDestination.setIsInbound(isInbound);
+        jmsDestination.setIsTemplate(isTemplate); // (kpak) - remove
+        jmsDestination.setProviderType(providerType);
+        jmsDestination.setInitialContextFactoryClassName(initialContextFactoryClassName);
+        jmsDestination.setJndiUrl(jndiUrl);
+        jmsDestination.setJndiUsername(jndiUsername);
+        jmsDestination.setJndiPassword(jndiPassword);
+        jmsDestination.setJndiProperties(jndiProperties);
+        
+        jmsDestination.setDestinationType(JmsDestination.DestinationType.fromType(destinationType));
+        jmsDestination.setConnectionFactoryName(connectionFactoryName);
+        jmsDestination.setDestinationName(destinationName);
+        jmsDestination.setDestinationUsername(destinationUsername);
+        jmsDestination.setDestinationPassword(destinationPassword);
+
+        jmsDestination.setInboundDetail(this.loadInboundDetail(jmsDestinationDetailsProps));
+        jmsDestination.setOutboundDetail(this.loadOutboundDetail(isTemplate, jmsDestinationDetailsProps, contextPropertiesTemplateProps));
+        
+        bundle.getJmsDestinations().put(name, jmsDestination);
+    }
+
+    private InboundJmsDestinationDetail loadInboundDetail(Map<String, Object> jmsDestinationDetailsProps) {
+        InboundJmsDestinationDetail inboundDetail = new InboundJmsDestinationDetail();
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Immediately return this expression instead of assigning it to the temporary variable "inboundDetail". ([squid:S1488](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS1488))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=138&issues=AWdc3146y5Us1olWmrcc&open=AWdc3146y5Us1olWmrcc)

[ISSUE_KEY]: AWdc3146y5Us1olWmrcc